src/backend/distributed/commands/index.c

```diff
@@ -442,23 +451,6 @@ PostprocessIndexStmt(Node *node, const char *queryString)
 	CommitTransactionCommand();
```

Comment:
Do we still need these commit/start calls? We'll do it anyway inside `ExecuteDistributedDDLJob` just before the distributed execution? I mean, we are not doing anything in between this `StartTransactionCommand ()` and `CommitTransactionCommand()` when `if (ddlJob->startNewTransaction)` is true?

Reply:
Yes, for the DROP INDEX CONCURRENTLY case.

Reply:
We also need to recommit after checking for metadata syncing, otherwise we still have a snapshot.